firewood/tests/db.rs

```diff
@@ -266,73 +231,50 @@ fn db_proposal() -> Result<(), DbError> {
         BatchOp::Delete { key: b"z" },
     ];
 
-    let proposal = Arc::new(db.new_proposal(batch)?);
-    let rev = proposal.get_revision();
-    assert_val!(rev, "k", "v");
+    let proposal = Arc::new(db.propose(batch).await?);
+    assert_val!(proposal, "k", "v");
 
     let batch_2 = vec![BatchOp::Put {
         key: b"k2",
         value: "v2".as_bytes().to_vec(),
     }];
-    let proposal_2 = proposal.clone().propose_sync(batch_2)?;
-    let rev = proposal_2.get_revision();
-    assert_val!(rev, "k", "v");
-    assert_val!(rev, "k2", "v2");
-
-    proposal.commit_sync()?;
-    proposal_2.commit_sync()?;
-
-    std::thread::scope(|scope| {
-        scope.spawn(|| -> Result<(), DbError> {
-            let another_batch = vec![BatchOp::Put {
-                key: b"another_k",
-                value: "another_v".as_bytes().to_vec(),
-            }];
-            let another_proposal = proposal.clone().propose_sync(another_batch)?;
-            let rev = another_proposal.get_revision();
-            assert_val!(rev, "k", "v");
-            assert_val!(rev, "another_k", "another_v");
-            // The proposal is invalid and cannot be committed
-            assert!(another_proposal.commit_sync().is_err());
-            Ok(())
-        });
-
-        scope.spawn(|| -> Result<(), DbError> {
-            let another_batch = vec![BatchOp::Put {
-                key: b"another_k_1",
-                value: "another_v_1".as_bytes().to_vec(),
-            }];
-            let another_proposal = proposal.clone().propose_sync(another_batch)?;
-            let rev = another_proposal.get_revision();
-            assert_val!(rev, "k", "v");
-            assert_val!(rev, "another_k_1", "another_v_1");
-            Ok(())
-        });
-    });
```

Comment:
I don't think that you should delete this; it's our only test that assures we can access nodes from multiple threads. While I think it's not really appropriately placed, it should remain as along as it's not properly replaced.

Reply:
Yes, except it's non-deterministic. For this test to actually catch something, it should be in a loop of multiple threads running simultaneously, not just a single lookup.

Reply:
It's not deterministic if things are being read in parallel, but the essence of the test still assures that you can at least attempt to do it. And we haven't had this test flake out, so it appears that parallel reads are actually possible. 

The async equivalent is as follows:
```rust
    let t1 = tokio::spawn({
        let proposal = proposal.clone();

        async move {
            let another_batch = vec![BatchOp::Put {
                key: b"another_k_t1",
                value: "another_v_t1".as_bytes().to_vec(),
            }];
            let another_proposal = proposal.clone().propose(another_batch).await.unwrap();
            let rev = another_proposal.get_revision();
            assert_val!(rev, "k", "v");
            assert_val!(rev, "another_k_t1", "another_v_t1");
            // The proposal is invalid and cannot be committed
            assert!(Arc::new(another_proposal).commit().await.is_err());
        }
    });

    let t2 = tokio::spawn({
        let proposal = proposal.clone();

        async move {
            let another_batch = vec![BatchOp::Put {
                key: b"another_k_t2",
                value: "another_v_t2".as_bytes().to_vec(),
            }];
            let another_proposal = proposal.clone().propose(another_batch).await.unwrap();
            let rev = another_proposal.get_revision();
            assert_val!(rev, "k", "v");
            assert_val!(rev, "another_k_t2", "another_v_t2");
        }
    });

    let _ = tokio::join!(t1, t2);
```